latest version + kyc artifact (#5)
casper-ecosystem/kyc-proxy-contract: README.md

# KYC Proxy Contract

> This is a proxy contract to check if an account is kyc proved through registered providers(KYC Contracts).

Reviewer: ```suggestion
This is a proxy contract to check if an account is KYC proved through registered providers (KYC Contracts).
```

)
```

> This proxy contract accepts a list of `contract_package_hash` on install deploy or singular package hashes on later deploys when calling the `"add_provider"` entrypoint.

Reviewer: ```suggestion
This proxy contract accepts a list of `contract_package_hash` on installation deploys or singular package hashes on later deploys when calling the `"add_provider"` entrypoint.
```

These providers can be banned or unbanned. Banned providers will not be asked for their opinion.

## Endpoints
### *init(initial_providers: Vec<ContractPackageHash>)*
Initialize proxy contracts with a given list of `contract_package_hash`

### *is_kyc_proved(account: Key, index: Option<U256>) -> bool*

> Check if a given account is kyc proved

Reviewer: ```suggestion
Check if a given account is KYC proved
```

### *add_kyc_provider(provider: Key)*

> Register a new kyc provider contract inside proxy contract

Reviewer: ```suggestion
Register a new KYC provider contract inside the proxy contract
```

### *ban_provider(provider: Key)*

> Set given kyc provider contract as validated inside proxy contract

Reviewer: ```suggestion
Set given KYC provider contract as validated inside the proxy contract
```

### *unban_provider(provider: Key)*

> Set given kyc provider contract as invalidated inside proxy contract

Reviewer: ```suggestion
Set given KYC provider contract as invalidated inside the proxy contract
```
